### PR TITLE
ENH: Styler.bar height control

### LINKED ISSUE #36419: ENH: I wish add a parameter to adjust bar function's height in Styler
#### Is your feature request related to a problem?

I have a 'Styler' object, I wish add a parameter that can adjust bar's spacing to the [bar function](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.io.formats.style.Styler.bar.html).

eg:
now the bar example like bottom:
image

but I wish like bottom:
image

Thx!!!

### PR DESCRIPTION
**Pull request by attack68**
- [x] closes #36419 
- [x] tests added / passed
- [x] whatsnew entry

I originally thought the issue was too complicated to code for such a small feature, but turns out it was only one or two lines of coding, so I added it in. Quite a nice addition for visualisations I think.

Screen Shot 2021-07-11 at 17 23 59

### CODE REVIEW

File: pandas/io/formats/style.py [hunk 1]
```diff
@@ -2911,7 +2922,13 @@ def css_calc(x, left: float, right: float, align: str):
             else:
                 start, end = z_frac, (x - left) / (right - left)
 
-        return css_bar(start * width, end * width, color)
+        ret = css_bar(start * width, end * width, color)
```
Comment (jreback) [inline pandas/io/formats/style.py:2925]: what happens for height / width out of bounds? do we raise, is this tested?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/io/formats/style.py (hunk 1 above)
Comment (attack68) [reply pandas/io/formats/style.py:2925]: currently still renders, but will get bad results. `height` is a css artifact though so bad CSS just means it gets ignored by the browser.

What do you want to do? Leave as is? Raise on out-of-bounds? or auto trim bad values to be in the range?

Comment (jreback) [reply pandas/io/formats/style.py:2925]: i think raising is appropriate (don't have to do here), prob needs a dedicated PR as likely need to audit all of these.

**Comment by jreback** [comment]
can you rebase

**Comment by attack68** [comment]
done, note the change to keyword only args.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks, IIRC elsewhere you are going to followon with a check for invalid height?

**Pull request merged**